arangodb/src/main/java/com/yahoo/ycsb/db/ArangoDBClient.java

```diff
@@ -0,0 +1,488 @@
+          recordcount, constructReturnForAQL(fields, "target"));
+
+      Map<String, Object> bindVars = new MapBuilder().put("key", startkey).get();
+      cursor = arangoDriver.executeDocumentQuery(aqlQuery, bindVars, null, BaseDocument.class);
+      Iterator<BaseDocument> iterator = cursor.entityIterator();
+      while (iterator.hasNext()) {
+        BaseDocument aDocument = iterator.next();
+        HashMap<String, ByteIterator> aMap = new HashMap<String, ByteIterator>(aDocument.getProperties().size());
+        this.fillMap(aMap, aDocument.getProperties());
+        result.add(aMap);
+      }
+      return Status.OK;
+    } catch (Exception e) {
+      logger.error("Exception while trying scan " + table + " " + startkey + " " + recordcount, e);
+    } finally {
+      if (cursor != null) {
+        try {
+          cursor.close();
+        } catch (ArangoException e) {
+          logger.error("Fail to close cursor", e);
+        }
+      }
+    }
+    return Status.ERROR;
+  }
+
+  private String createDocumentHandle(String collectionName, String documentKey) throws ArangoException {
+    validateCollectionName(collectionName);
+    return collectionName + "/" + documentKey;
+  }
+
+  private void validateCollectionName(String name) throws ArangoException {
+    if (name.indexOf('/') != -1) {
+      throw new ArangoException("does not allow '/' in name.");
+    }
+  }
+
+  
+  private String constructReturnForAQL(Set<String> fields, String targetName) {
+    // Construct the AQL query string.
+    String resultDes = targetName;
+    if (fields != null && fields.size() != 0) {
+      StringBuilder builder = new StringBuilder("{");
+      for (String field : fields) {
+        builder.append(String.format("\n\"%s\" : %s.%s,", field, targetName, field));
+      }
+      //Replace last ',' to newline.
+      builder.setCharAt(builder.length() - 1, '\n');
+      builder.append("}");
+      resultDes = builder.toString();
+    }
+    return resultDes;
+  }
+  
+
+  private void fillMap(Map<String, ByteIterator> resultMap, Map<String, Object> properties) {
+    fillMap(resultMap, properties, null);
+  }
+  
+  /**
+   * Fills the map with the properties from the BaseDocument.
+   * 
+   * @param resultMap
+   *      The map to fill/
+   * @param obj
+   *      The object to copy values from.
+   */
+  @SuppressWarnings("unchecked")
+  private void fillMap(Map<String, ByteIterator> resultMap, Map<String, Object> properties, Set<String> fields) {
+    if (fields == null || fields.size() == 0) {
+      for (Map.Entry<String, Object> entry : properties.entrySet()) {
+        if (entry.getValue() instanceof String) {
+          resultMap.put(entry.getKey(),
+              stringToByteIterator((String)(entry.getValue())));
+        } else {
+          logger.error("Error! Not the format expected! Actually is {}",
+              entry.getValue().getClass().getName());
+          System.exit(-1);
```

Comment:
Should never call System.exit. Please find a way to propagate the error instead.